core/imds/src/main/java/software/amazon/awssdk/imds/internal/DefaultEc2Metadata.java

```diff
@@ -86,191 +88,110 @@ public Ec2Metadata.Builder toBuilder() {
 
-        try {
-            long backoffTimeMillis = getBackoffDuration(tries);
-            Thread.sleep(backoffTimeMillis);
-        } catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-            throw SdkClientException.builder().message("Thread interrupted while trying to sleep").cause(e).build();
+        SdkClientException.Builder sdkClientExceptionBuilder = SdkClientException
+            .builder()
+            .message("Exceeded maximum number of retries. Total attempts: " + retryPolicy.numRetries() + ".");
+        if (lastCause != null) {
+            String msg = sdkClientExceptionBuilder.message() + " " + lastCause.getMessage();
+            sdkClientExceptionBuilder.cause(lastCause).message(msg);
         }
+        throw sdkClientExceptionBuilder.build();
     }
 
-    private HttpExecuteResponse getDataHttpResponse(String path, String token) throws IOException {
+    private MetadataResponse doRequest(String path, String token) throws IOException {
 
-        URI uri = URI.create(endpoint + path);
-        HttpExecuteRequest httpExecuteRequest = REQUEST_MARSHALLER.createDataRequest(uri, SdkHttpMethod.GET, token,
-                                                                                     tokenTtl);
-        return httpClient.prepareRequest(httpExecuteRequest).call();
-    }
+        HttpExecuteRequest httpExecuteRequest = requestMarshaller.createDataRequest(path, token, tokenTtl);
+        HttpExecuteResponse response = httpClient.prepareRequest(httpExecuteRequest).call();
 
-    private long getBackoffDuration(int tries) {
+        int statusCode = response.httpResponse().statusCode();
+        Optional<AbortableInputStream> responseBody = response.responseBody();
 
-        long backoffTime = 0L;
-        if (tries >= 1) {
-            backoffTime = retryPolicy.backoffStrategy()
-                                              .computeDelayBeforeNextRetry(RetryPolicyContext.builder()
-                                                                                             .retriesAttempted(tries - 1)
-                                                                                             .build()).toMillis();
+        if (!HttpStatusFamily.of(statusCode).isOneOf(HttpStatusFamily.SUCCESSFUL)) {
+            responseBody.map(this::uncheckedInputStreamToUtf8)
+                        .ifPresent(str -> log.debug(() -> "Metadata request response body: " + str));
+            throw SdkClientException.builder()
+                        .message("The requested metadata at path ( " + path + " ) return Http code " + statusCode).build();
         }
 
-        return backoffTime;
+        AbortableInputStream abortableInputStream = responseBody.orElseThrow(
+            SdkClientException.builder().message("Response body empty with Status Code " + statusCode)::build);
+        String data = uncheckedInputStreamToUtf8(abortableInputStream);
+        return new MetadataResponse(data);
```

Comment:
Can we create static factory method? The naming convention is `create`.  https://github.com/aws/aws-sdk-java-v2/blob/master/docs/design/FavorStaticFactoryMethods.md#naming-conventions

Reply:
ack